structuredheadings/plugin.js

```diff
@@ -18,8 +55,31 @@
   var firstHeaderKey = 0;
   var lastHeaderKey = headerList.length - 1;
 
+//style helpers
+
+
+  var clearStyles = function (element) {
+    for (var styleName in config.autonumberStyles) {
+      var style = config.autonumberStyles[styleName];
+      for (var className in style) {
+        element.removeClass(style[className]);
+      }
+    }
+  };
+
+  var setStyle = function (element, styleName) {
+    var elementType = element.getName();
+    var style = config.autonumberStyles[styleName];
+    if (elementType) {
```

Comment:
Yes, otherwise the reapplyStyle (which loops over the entire document) chokes. But it might be better to use if(element.type === CKEDITOR.NODE_ELEMENT)